### DISCUSSION TITLE
file structure

### DISCUSSION THREAD

**Discussion by AmbassadorWEB3** [EtWnn/ScanWatch#34] [General]
I found your program and installed it, unfortunately I am just beginning to understand how Python works, because I studied Pascal and Delphi back in 1998, but then I started working with servers and gave up programming. Now I wanted to get back into programming so I'm studying Solidity. 
Your program is an interesting tool, but unfortunately I ran into a problem: I can't figure out how/where I should put the data from section 3. Manager - I understood correctly that it should be the file Manager.py with the code?:

```
from ScanWatch.ScanManager import ScanManager
from ScanWatch.utils.enums import NETWORK
api_token = "<ETH_API_TOKEN>"  # I understand that here I have to substitute my API key 
address = "<YOUR_ETH_ADDRESS>" # I understand that here I have to substitute my ETH address 
manager = ScanManager(address, NETWORK.ETHER, api_token)
```

Then what should partition 4 look like? Is it a *.py file, too? What about section 5? 6?
**Please help me understand the file structure, if possible send me a screenshot of your files or write me a more detailed explanation of what I have to do?**
I will be very grateful to you!

**Comment by EtWnn** [comment]
Hi @AmbassadorWEB3,

Welcome to the python world! You will see that it is one of the programming language with the most developped library ecosystem.
I am glad that that you have found some utility in this small tool that I shared here 😁

Here is a small step by step tutorial for you.

## Anaconda installation

To manage differents environnements (ide different python version and/or potentially conflicting libraries) most begginners tends to use Anaconda as it is simple to use.
You can install it here: https://docs.anaconda.com/anaconda/install/index.html

## Environement creation

to create a new environment that we will name scanwatch_env, enter the command in a terminal:

`conda create -n scanwatch_env python=3.8 -y`

to use this environment,  you will need to activate it every time you open a new terminal. Let's do that now:

`conda activate scanwatch_env`

You should see the name of the environment on the left of your terminal.

## ScanWatch intstallation

Install the package with pip:

`pip install -U scanwatch`

## Jupyter installation

To play around with the data, I would recommand using a jupyter notebook, it would be easier to begin with.
To install it, type: 

`conda install jupyter -y`

## Package utilisation

Copy the file shared at the bottom of this message, uncompress and past it in a folder of your choice.
in your terminal, use the `cd` command to go to this folder and then start jupyter by entering in a terminal:

`jupyter-notebook`

After some wait time, a browser page will open and you should see your file.
You can execute the cells one by one and see what happens.
I'd recommand you to follow a short jupyter notebook tutorial to get the gist of what's going on

And that's it! Don't hesitate to ask me other question if you need

## Note

If you want to execute my package in a *.py file,  there is also an example in the zip file and you can execute it like this

`python scan_watch_example.py`

[scan_watch_example.zip](https://github.com/EtWnn/ScanWatch/files/11075775/scan_watch_example.zip)
